overlord/backend.go

```diff
@@ -33,7 +33,7 @@ type overlordStateBackend struct {
 }
 
 func (osb *overlordStateBackend) Checkpoint(data []byte) error {
-	return osutil.AtomicWriteFile(osb.path, data, 0600, 0)
+	return osutil.AtomicWriteFile(osb.path, data, 0600, osutil.AtomicWriteImmutable)
```

Comment:
It would be good to know if the FS actually supports this. I think we're fine for now (ext4) but then there's windows and other Ones and what not.